src/desktop/DesktopUtils.js

```diff
@@ -142,6 +142,62 @@ export default class DesktopUtils {
 				return Promise.reject(new Error(`invalid platform: ${process.platform}`))
 		}
 	}
+
+	/**
+	 * reads the lockfile and then writes the own version into the lockfile
+	 * @returns {*} whether the lock was overridden by another version
+	 */
+	static singleInstanceLockOverridden(): Promise<boolean> {
+		const lockfilePath = getLockFilePath()
+		return fs.readFile(lockfilePath, 'utf8')
```

Comment:
Does this assume that the file is there and can be read?

Reply:
Do I understand correctly that the second instance with different version wins? It doesn't compare if it's bigger/smaller?
Maybe it's okay but I would think about it again.

Reply:
Yes, it allows older versions to steal the lock from newer ones. It's assumed that users that start a version actually want to run that version and makes it symmetric when you're using different versions (no surprise when one version can override the other one but not vice versa).

The lockfile is assumed to be there, since the app creates it when it's started. It should probably fall back to the old behaviour (version agnostic single instance lock) when the file is unavailable.

Reply:
I think the golden rule is "expect any IO to fail" and because this one is at startup and so essential, I would at least put `catch` there.
I think overriding with older version is fine, it will be updated anyway.